File: JAVA/level2/프린터.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        int index = priorities.length-1;
        Queue<Integer> queue = new LinkedList<Integer>();

        for(int i=0; i<priorities.length; i++){
            if(location == i)
                queue.offer(priorities[i]*-1);
            else
                queue.offer(priorities[i]);
        }

        Arrays.sort(priorities);

        while(!queue.isEmpty()){
            if(queue.peek() == priorities[index] || queue.peek()*-1 == priorities[index]){
                if(queue.peek() < 0)
                    return answer+1;
                queue.poll();
                answer++;
                index--;
            }
            else{
                int temp = queue.peek();
                queue.poll();
                queue.offer(temp);
            }
        }

        return answer;
    }
}

File: JavaScript/level2/프린터.js
function compareNumbers(a, b) {
    return b - a;
}

function solution(priorities, location) {
    var answer = 0;
    let order = priorities.map((val, index) => [val, index]);

    priorities.sort(compareNumbers);
    
    while(order.length !== 0){
        if(order[0][0] === priorities[0]){
            answer++;
            if(order[0][1] === location)
                break;
            order.shift();
            priorities.shift();
        }
        else{
            let temp = order.shift();
            order.push(temp);
        }
    }
    
    return answer;
}